tests/testthat/test-CRANSKIP-hooks.R

```diff
@@ -1,36 +1,39 @@
-library(logger)
-library(testthat)
+eval_outside <- function(...) {
 
-eval_outside <- function(expr) {
-    t <- tempfile()
-    on.exit(unlink(t))
-    cat('library(logger); log_messages(); log_warnings(); log_errors();', file = t)
-    cat(expr, file = t, append = TRUE, sep = '\n')
-    paste(
-        suppressWarnings(system(paste('$R_HOME/bin/Rscript', t, '2>&1'), intern = TRUE)),
-        collapse = '\n')
+    input <- normalizePath(withr::local_tempfile(lines = character()), winslash = "/")
+    output <- normalizePath(withr::local_tempfile(lines = character()), winslash = "/")
+    writeLines(con = input, c(
+        "library(logger)",
+        "log_layout(layout_glue_generator('{level} {msg}'))",
+        paste0("log_appender(appender_file('", output, "'))"),
+        ...
+    ))
+    path <- file.path(R.home("bin"), "Rscript")
+    if (Sys.info()[["sysname"]] == "Windows") {
+        path <- paste0(path, ".exe")
+    }
+    suppressWarnings(system2(path, input, stdout = TRUE, stderr = TRUE))
+    readLines(output)
 }
 
 test_that('log_messages', {
-    expect_match(eval_outside('message(42)'), 'INFO')
-    if (R.Version()$os == 'linux-gnu') {
-        expect_match(eval_outside('system("echo 42", invisible = TRUE)'), 'INFO')
-    }
+    expect_snapshot({
+        writeLines(eval_outside("log_messages()", 'message(42)'))
```

Comment:
Similarly I moved all these to snapshot tests, which makes it easier to see what's going wrong if there are failures.